src/main/java/org/jabref/gui/bibtexextractor/BibtexExtractorViewModel.java

```diff
@@ -1,32 +1,99 @@
 import org.jabref.model.entry.types.EntryType;
 import org.jabref.model.entry.types.StandardEntryType;
+import org.jabref.preferences.JabRefPreferences;
 
 public class BibtexExtractorViewModel {
 
     private final StringProperty inputTextProperty = new SimpleStringProperty("");
     private final BibDatabaseContext bibdatabaseContext;
+    private List<BibEntry> extractedEntries;
+    private final BibDatabaseContext newDatabaseContext;
+    private boolean directAdd;
+    private DialogService dialogService;
+    private GrobidCitationFetcher currentCitationfetcher;
 
     public BibtexExtractorViewModel(BibDatabaseContext bibdatabaseContext) {
         this.bibdatabaseContext = bibdatabaseContext;
+        newDatabaseContext = new BibDatabaseContext(new Defaults(BibDatabaseMode.BIBTEX));
+        dialogService = JabRefGUI.getMainFrame().getDialogService();
     }
 
     public StringProperty inputTextProperty() {
         return this.inputTextProperty;
     }
 
-    public void startExtraction() {
+    public void startParsing(boolean directAdd) {
+        this.directAdd = directAdd;
+        this.extractedEntries = null;
+        Task<Void> parseUsingGrobid = new Task<Void>() {
+            @Override
+            protected Void call() throws Exception {
+                try {
+                    currentCitationfetcher = new GrobidCitationFetcher(
+                        JabRefPreferences.getInstance().getImportFormatPreferences(),
+                        Globals.getFileUpdateMonitor(),
+                        Globals.prefs
+                    );
+                    extractedEntries = currentCitationfetcher.performSearch(inputTextProperty.getValue());
+                } catch (FetcherException e) {
+                    extractedEntries = new ArrayList<>();
+                }
+                Platform.runLater(() -> executeParse());
+                return null;
+            }
+        };
+        dialogService.showProgressDialogAndWait(Localization.lang("Your text is being parsed.."),Localization.lang( "Please wait while we are parsing your text"), parseUsingGrobid);
+        Globals.TASK_EXECUTOR.execute(parseUsingGrobid);
+    }
 
+    public void executeParse() {
+        if (extractedEntries.size() > 0) {
+            if (directAdd) {
+                BibtexKeyGenerator bibtexKeyGenerator = new BibtexKeyGenerator(newDatabaseContext, Globals.prefs.getBibtexKeyPatternPreferences());
+                for (BibEntry bibEntries: extractedEntries) {
+                    bibtexKeyGenerator.generateAndSetKey(bibEntries);
+                    newDatabaseContext.getDatabase().insertEntry(bibEntries);
+                }
+                newDatabaseContext.setMode(BibDatabaseMode.BIBTEX);
+                JabRefGUI.getMainFrame().addTab(newDatabaseContext,true);
+            } else {
+                for (BibEntry bibEntry: extractedEntries) {
+                    this.bibdatabaseContext.getDatabase().insertEntry(bibEntry);
```

Comment:
Please use https://github.com/JabRef/jabref/blob/381f0f3f0876a953e274606027a8e2be0633f10d/src/main/java/org/jabref/gui/externalfiles/ImportHandler.java#L110 to add the entries, which also takes care of a few other things.